### DISCUSSION TITLE
Add `BlockApplyEffectsEvent`

### DISCUSSION THREAD

**Discussion by cometcake575** [PaperMC/Paper#10517] [Ideas]
### Is your feature request related to a problem?

Yes, I am trying to apply potion effects to players outside of water when there is a nearby conduit in water, but currently there is no good way to test for an active conduit trying to give effects.

### Describe the solution you'd like.

A `BlockApplyEffects` event, for beacons and conduits, containing a `getBlock()` method and a `getPower()` method for the size of the conduit prismarine structure or beacon pyramid.

This event would be called when a beacon or conduit is about to iterate over players to apply effects, firing once each time regardless of if no players are going to be given the effect.

### Describe alternatives you've considered.

Looping through nearby blocks for an active conduit or beacon, however this would not work as it would have to often loop through large amounts of blocks, likely causing the server to crash or slow down by a lot as it wouldn't be fast enough to handle searching through so many nearby blocks

### Other

Another name for the event might be better, as `BlockApplyEffects` may suggest that effects are actually being applied rather than the game being about to try applying them, I suggested it with that name since the method for trying to apply effects in the game source code is `applyEffects()`

**Comment by Machine-Maker** [comment]
I like the concept for this event. I think the name probably needs some work to further distinguish it from BeaconEffectEvent which is fired for each player an effect is being applied to for beacons.

We have a similar PR open for spawners adding `SpawnerPlayerSearchEvent`, allowing plugins to change how spawners detect players in range and I think a similar approach for this would be good.
